tfx_bsl/tfxio/parquet_tfxio.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TFXIO implementation for Parquet."""
+
+import copy
+from typing import Optional, List, Text, Any
+
+import apache_beam as beam
+import pyarrow as pa
+import tensorflow as tf
+from tensorflow_metadata.proto.v0 import schema_pb2
+
+from tfx_bsl.coders import csv_decoder
+
+from tfx_bsl.tfxio import dataset_options, tensor_adapter, tensor_representation_util, telemetry
+from tfx_bsl.tfxio.tfxio import TFXIO
+
+PARQUET_FORMAT = "parquet"
+
+
+class ParquetTFXIO(TFXIO):
+  """TFXIO implementation for Parquet."""
+
+  def __init__(self,
+               file_pattern: Text,
+               column_names: List[Text],
+               min_bundle_size: int = 0,
+               schema: Optional[schema_pb2.Schema] = None,
```

Comment:
can parquet store other data types apart from dense, e.g. varying length features?
In other words, would it be possible to infer schema automatically if not provided?

Reply:
yeah, it can be inferred. If you look at the tests, there's one without specifying a schema. The missing part if inferring the schema. Do you think that is something we want to do? The inference will need to happen by reading one of the parquet files.

Reply:
@iindyk done in https://github.com/tensorflow/tfx-bsl/pull/52/commits/217cb9c2eb0340b8b2a91434b38348fafceb0975